cmd/minikube/cmd/config/set_test.go

```diff
@@ -19,68 +19,62 @@ package config
 import (
 	"io/ioutil"
 	"os"
-	"path/filepath"
 	"testing"
 
-	"k8s.io/minikube/pkg/minikube/config"
 	"k8s.io/minikube/pkg/minikube/localpath"
 )
 
 func TestNotFound(t *testing.T) {
-	createTestProfile(t)
+	createTestConfig(t)
 	err := Set("nonexistent", "10")
-	if err == nil {
+	if err == nil || err.Error() != "property name \"nonexistent\" not found" {
 		t.Fatalf("Set did not return error for unknown property")
 	}
 }
 
 func TestSetNotAllowed(t *testing.T) {
-	createTestProfile(t)
+	createTestConfig(t)
 	err := Set("vm-driver", "123456")
 	if err == nil || err.Error() != "[driver \"123456\" is not supported]" {
 		t.Fatalf("Set did not return error for unallowed value")
 	}
 }
 
 func TestSetOK(t *testing.T) {
-	createTestProfile(t)
+	createTestConfig(t)
 	err := Set("vm-driver", "virtualbox")
 	defer func() {
 		err = Unset("vm-driver")
 		if err != nil {
-			t.Errorf("failed to unset vm-driver")
+			t.Errorf("failed to unset vm-driver: %+v", err)
 		}
 	}()
 	if err != nil {
-		t.Fatalf("Set returned error for valid property value")
+		t.Fatalf("Set returned error for valid property value: %+v", err)
 	}
 	val, err := Get("vm-driver")
 	if err != nil {
-		t.Fatalf("Get returned error for valid property")
+		t.Fatalf("Get returned error for valid property: %+v", err)
 	}
 	if val != "virtualbox" {
 		t.Fatalf("Get returned %s, expected \"virtualbox\"", val)
 	}
 }
 
-func createTestProfile(t *testing.T) {
+func createTestConfig(t *testing.T) {
```

Comment:
config.Get/Set/Unset functions are reading the values from config/config.json only. So creating temp profile here doesn't make sense. 

I have renamed the function and change implementation to create team config.